frontend/src/views/DashboardPage/DashboardPage.tsx

```diff
@@ -734,6 +749,54 @@ export const DashboardPage = () => {
   const isSecretImportEmpty = !secretImportCfg?.imports?.length;
   const isEmptyPage = isFoldersEmpty && isSecretEmpty && isSecretImportEmpty;
 
+
+  const [selectedTags, setSelectedtags] = useState<WsTag[]>([])
+  const [selectedTagIds, setSelectedTagIds] = useState<Record<string, boolean>>({})
+  const [hoveredTag, setHoveredTag] = useState<WsTag | null>(null);
+
+  const handleTagOnMouseEnter = (wsTag: WsTag) => {
+    setHoveredTag(wsTag);
+  };
+
+  const handleTagOnMouseLeave = () => {
+    setHoveredTag(null);
+  };
+
+  const checkIfTagIsVisible = (wsTag: WsTag) => wsTag._id === hoveredTag?._id;
+
+  useEffect(() => {
+    const secCheckBox = document.querySelector("#sec-checkbox")
```

Comment:
Kindly avoid this pattern. Use react ref and not querySelector. Please change this part. 

And you don't need to use useEffect + querySelector like this. Just do a statue check and do a direct check inside the document node itself.

Reply:
Okay, makes sense!

Reply:
Addressed!